api/proto/teleport/externalaudit/v1/externalaudit_service.proto

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package teleport.externalaudit.v1;
+
+import "google/protobuf/empty.proto";
+import "teleport/externalaudit/v1/externalaudit.proto";
+
+option go_package = "github.com/gravitational/teleport/api/gen/proto/go/teleport/externalaudit/v1;externalauditv1";
+
+// ExternalAuditService provides methods to manage external audit.
+// It's used only in Teleport Cloud for feature called "bring your own audit bucket".
+//
+// There could be multiple external audit resources in cluster, but only single
+// resource can be set as 'Active'.
+// The one which is active is used by auth.
+//
+// Supporting multiple ExternalAudit as resources allow us to to add resource,
+// and test connection before actually enabling it in cluster.
+service ExternalAuditService {
+  // GetExternalAudit returns the external audit configuration resource.
+  rpc GetExternalAudit(GetExternalAuditRequest) returns (GetExternalAuditResponse);
+
+  // CreateExternalAudit creates a new External Audit resource.
+  // ExternalAudit after creation is set to 'draft' status.
+  rpc CreateExternalAudit(CreateExternalAuditRequest) returns (CreateExternalAuditResponse);
+
+  // DeleteExternalAudit removes External Audit resource.
+  // You cannot remove external audit that is set as 'active'.
+  // First you need to SetExternalAuditStatus to `draft` on that resource.
+  rpc DeleteExternalAudit(DeleteExternalAuditRequest) returns (google.protobuf.Empty);
+
+  // SetExternalAuditStatus sets external audit status based on name of external audit.
+  // Setting to draft->active or active->draft will cause service reload.
+  // Only single resource can be set to active state.
+  rpc SetExternalAuditStatus(SetExternalAuditStatusRequest) returns (SetExternalAuditStatusResponse);
```

Comment:
Can we change between configs?
That means we would get 0 active or 2 active, which does not make sense I think.
That's why I was suggesting to have this resource being referenced by another global setting

Reply:
I have removed that endpoint + status and added new `SetClusterExternalAudit`. It should make more sense now.